ExeterTeam25/group-software-project
# Challenge expiry

WARNING: Make sure to run makemigrations and migrate before you run this!

To test creation:
- Run the server and log in as privileged user
- Go to create a challenge
- Attempt to set expiry date to before current date (It shouldn't let you)

To test display:
- Create a challenge with a valid expiry date
- Log into admin portal
- Edit said challenge to expire before now
- Go back home
- Ensure that challenge isn't displayed in any way (either marker or list)

To test completion:
- Go to the complete challenge screen of the previously edited challenge (/create-response/{challengeid}/)
- Attempt to submit response
- You should be redirected to home and an error message is displayed

To test multiple submission fix:
- Go to the complete challenge screen of a previously completed challenge (Expired or not) (/create-response/{challengeid}/)
- You should be redirected to home with an error message

Details:
Added expiry date + time in challenge creation
Added features preventing users from completing challenges twice and completing expired challenges
Made it so only unexpired challenges are displayed
Challenges must finish after creation date

TODO:
Use this code to implement weekly challenges
